src/rgw/rgw_rest_bucket.cc

```diff
@@ -219,7 +219,7 @@ void RGWOp_Bucket_Remove::execute(optional_yield y)
   RESTArgs::get_string(s, "bucket", bucket_name, &bucket_name);
   RESTArgs::get_bool(s, "purge-objects", false, &delete_children);
 
-  op_ret = store->get_bucket(s, nullptr, string(), bucket_name, &bucket, y);
+  op_ret = store->get_bucket(s, s->user.get(), string(), bucket_name, &bucket, y);
```

Comment:
this will store `s->user` in `sal::Bucket::owner` - but `s->user` isn't the owner, it's the user that sent the `RGWOp_Bucket_Remove` request

does it make sense for `Store::get_bucket()` to take a pointer to the owner? we generally don't know the bucket's owner before loading the bucket

is there a better way to stop crashing here than passing a fake user?

Reply:
I don't know.  This is what was done before zipper, so I thought the simplest solution was to make it work like it used to work.  Is there some way to get the actual owner?

Reply:
> This is what was done before zipper, so I thought the simplest solution was to make it work like it used to work.

we didn't do anything here before zipper (https://github.com/ceph/ceph/blob/octopus/src/rgw/rgw_rest_bucket.cc#L216-L238), we just called `forward_request_to_master()` directly

> Is there some way to get the actual owner?

you'd have to load the bucket info from rados, which we shouldn't need to do

Reply:
i think the issue is in `RadosBucket::remove_bucket()`, which uses `owner` for the forwarded request:
```c++
ret = store->forward_request_to_master(dpp, owner, &ot.read_version, in_data, nullptr, *req_info, y);
```
that should be `s->user` in this case

Reply:
which is, in a round-about way, what this does.  We don't have access to s here (by design).  We could add a user to the call, I suppose, but there's already one in the bucket, so I just used that.

Reply:
ok, if you don't want to change interfaces, can we at least add a FIXME comment to explain that we're misusing the bucket's owner for `forward_request_to_master()`?